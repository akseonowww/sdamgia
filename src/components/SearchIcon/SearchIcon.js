import React from 'react';

import './SearchIcon.scss';

const SearchIcon = ({ className }) => (
	<div className={`SearchIcon ${className}`}>
		<svg
			className="SearchIcon-Svg"
			version="1.1"
			viewBox="0 0 27.9 50"
			x="0px"
			y="0px"
			xmlns="http://www.w3.org/2000/svg"
		>
			<circle
				className="SearchIcon-Lens"
				cx="10.9"
				cy="24.1"
				r="9.4"
			></circle>
			<rect
				className="SearchIcon-Handle"
				width="3"
				height="14.5"
				x="20"
				y="27.6"
				transform="matrix(0.7071 -0.7071 0.7071 0.7071 -18.3066 25.467)"
			></rect>
		</svg>
	</div>
);

export default SearchIcon;
