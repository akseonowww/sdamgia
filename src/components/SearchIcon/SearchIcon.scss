@import "src/styles/variables";

.SearchIcon {
  width: 14px;

  &-Svg {
    enable-background: new 0 0 27.9 50;
  }

  &-Lens {
    fill: none;
    stroke: $primary;
    stroke-width: 3;
    stroke-miterlimit: 10;
  }

  &-Handle {
    fill: $primary;
  }
}
