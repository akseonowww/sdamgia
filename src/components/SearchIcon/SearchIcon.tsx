import React, { FC } from 'react'
import { cn } from '@bem-react/classname'
import { classnames as cx } from '@bem-react/classnames'

import './SearchIcon.scss'

interface ISearchIconProps {
  className?: string
}

const SearchIcon: FC<ISearchIconProps> = ({ className }) => {
  const searchIcon = cn('SearchIcon')

  return (
    <div className={cx(searchIcon(), className)}>
      <svg
        className={searchIcon('Svg')}
        version="1.1"
        viewBox="0 0 27.9 50"
        x="0px"
        y="0px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          className={searchIcon('Lens')}
          cx="10.9"
          cy="24.1"
          r="9.4"
        ></circle>
        <rect
          className={searchIcon('Handle')}
          width="3"
          height="14.5"
          x="20"
          y="27.6"
          transform="matrix(0.7071 -0.7071 0.7071 0.7071 -18.3066 25.467)"
        ></rect>
      </svg>
    </div>
  )
}

export default SearchIcon
