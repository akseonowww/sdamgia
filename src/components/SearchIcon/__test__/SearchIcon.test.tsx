import React from 'react'
import { unmountComponentAtNode, render } from 'react-dom'
import { act } from 'react-dom/test-utils'
import pretty from 'pretty'

import SearchIcon from '../SearchIcon'

describe('SearchIcon', () => {
  let container: HTMLElement | null = null

  beforeEach(() => {
    container = document.createElement('div')
    document.body.appendChild(container)
  })

  afterEach(() => {
    if (container) {
      unmountComponentAtNode(container)
      container.remove()
      container = null
    }
  })

  it('should render an icon', () => {
    act(() => {
      render(<SearchIcon />, container)
    })
    if (container) expect(pretty(container.outerHTML)).toMatchSnapshot()
  })
})
