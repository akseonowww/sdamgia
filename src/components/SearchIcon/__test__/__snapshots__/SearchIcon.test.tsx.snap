// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchIcon should render an icon 1`] = `
"<div class=\\"SearchIcon\\"><svg class=\\"SearchIcon-Svg\\" version=\\"1.1\\" viewBox=\\"0 0 27.9 50\\" x=\\"0px\\" y=\\"0px\\" xmlns=\\"http://www.w3.org/2000/svg\\">
    <circle class=\\"SearchIcon-Lens\\" cx=\\"10.9\\" cy=\\"24.1\\" r=\\"9.4\\"></circle>
    <rect class=\\"SearchIcon-Handle\\" width=\\"3\\" height=\\"14.5\\" x=\\"20\\" y=\\"27.6\\" transform=\\"matrix(0.7071 -0.7071 0.7071 0.7071 -18.3066 25.467)\\"></rect>
  </svg></div>"
`;
